src/common/robot.rs

```diff
@@ -130,14 +143,18 @@ impl LocalRobot {
             resources: ResourceMap::new(),
         };
 
-        let components = &(config_resp.config.as_ref().unwrap().components);
+        let mut components = config_resp.config.as_ref().unwrap().components.clone();
```

Comment:
is there a way we can avoid cloning here? it doesn't look like you are keeping ownership of the cloned components

Reply:
I see now why you need it. Do you think we could write the algorithm so that the vector you allocate is a Vec<Option<&ComponentConfig>> this way as you can just take the inner value of the Option and put it back if it needs further processing. Otherwise the Vector will contently move things around in memory which can be problemactic for larger configs. Could also use a Map here. If this is too complicated then I think we should do a Vec<&ComponentConfig> so it takes less space in memory